### PR TITLE
Displays application output in a separate window - GDB tty / New-Console

### PR DESCRIPTION
**Pull request by marcsosduma**
Hello! Oleg and Simon,

as you indicated, I changed the code to allow debugging in a separate window. I did the tests and it works on Windows and Linux.

Thanks,

Duma

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,6 +164,35 @@ Here's an example:
 }
 ```
 
+### Displays application output in a separate window
+Add `gdbtty` property to your `lanch.json`. Here’s an example:
```
Comment (OlegKunitsyn) [inline README.md:168]: a typo 'lanch'

**Review by OlegKunitsyn — APPROVED** [review]

**Comment by GitMensch** [comment]
Thank you for the implementation of this. As a follow-up: it should also be possible (at least on GNU/Linux, not sure about Win32) to create a terminal within vscode, then use its tty instead of the xterm one.

@marcsosduma Did you inspected that option or can do so, please (as an alternative to xterm)?

**Comment by marcsosduma** [comment]
Hi Simon! I got to read a bit about xterm.js. But if it's possible to use
vscode's own editor it looks better. I'll do some research. Thanks for the
tip (sorry for bad English).

Em dom., 9 de jul. de 2023 às 12:13, Simon Sobisch ***@***.***>
escreveu:

**Comment by GitMensch** [comment]
Your English is fine (if unsure I'd suggest to use deepl.com for both [translation](https://www.deepl.com/translator) and [language check](https://www.deepl.com/write)).

If you can implement the internal terminal then it is most likely reasonable to change the type of `gdbtty` to an enum: `false, true, internal, external` (with `true`, included for compatibility being one of the later [I'd suggest "internal"]).

**Comment by marcsosduma** [comment]
Hello Simon and Oleg!
I found these examples:
https://github.com/microsoft/vscode-extension-samples
I'll study and see if I can understand how they work.

Em seg., 10 de jul. de 2023 às 04:20, Simon Sobisch <
***@***.***> escreveu:

**Comment by marcsosduma** [comment]
Hello Simon and Oleg!

do you think it's really bad to add the parameter "gdbttytype" with the
options: xterm / vscode ?

example:
     {
            "name": "COBOL debugger",
            "type": "gdb",
            "request": "launch",
            "preLaunchTask": "cobol build",
            "cobcargs": [
                "-free",
                "-x",
                "-lncursesw"
            ],
            "coverage": false,
            "gdbtty":true,
          *  "gdbttytype": "vscode"*
        }


[image: image.png]

Em qui., 13 de jul. de 2023 às 14:49, Marcos Duma ***@***.***>
escreveu:

**Pull request merged**